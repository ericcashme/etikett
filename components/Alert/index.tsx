import React, { FC, memo, useEffect, useState, useCallback } from 'react'
import { Button } from '../Button'
import * as S from './styles'
import { AlertProps } from './types'

export const Alert: FC<AlertProps> = memo(
  ({
    size = 'medium',
    close,
    action,
    children,
    isLoading = false,
    isDisabled = false,
    cancelButtonName = 'Cancel',
    confirmButtonName = 'Confirm'
  }) => {
    const [body] = useState(document?.querySelector('body') as HTMLBodyElement)

    const handleKeyUp = useCallback(
      (e) => {
        if (e.keyCode === 27) {
          e.preventDefault()
          close()
          window.removeEventListener('keyup', handleKeyUp, false)
        }
      },
      [close]
    )

    useEffect(() => {
      window.addEventListener('keyup', handleKeyUp, false)
      body.style.overflow = 'hidden'

      return () => {
        window.removeEventListener('keyup', handleKeyUp, false)
        body.style.overflow = 'visible'
      }
    }, [handleKeyUp, body])

    return (
      <S.Background>
        <S.Content size={size}>
          <S.InnerContent>{children}</S.InnerContent>

          <S.Actions>
            <Button kind="weak" isOutlined onClick={close}>
              {confirmButtonName}
            </Button>

            <Button kind="warning" isLoading={isLoading} isDisabled={isDisabled || isLoading} onClick={action}>
              {cancelButtonName}
            </Button>
          </S.Actions>
        </S.Content>
      </S.Background>
    )
  }
)

Alert.displayName = 'Alert'
