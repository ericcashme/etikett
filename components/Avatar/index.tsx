import React, { FC, memo } from 'react'
import Image from 'next/image'
import * as S from './styles'
import { AvatarProps } from './types'

export const Avatar: FC<AvatarProps> = memo(({ size = 'md', image, alt, hasBorder = false, ...rest }) => (
  <S.Wrap size={size} hasBorder={hasBorder} {...rest}>
    <Image src={image} alt={alt} width={S.getSize[size]} height={S.getSize[size]} layout="responsive" />
  </S.Wrap>
))

Avatar.displayName = 'Avatar'
