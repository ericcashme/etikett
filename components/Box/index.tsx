import React, { FC, memo } from 'react'
import * as S from './styles'
import { BoxProps } from './types'

export const Box: FC<BoxProps> = memo(
  ({
    type = 'default',
    kind = 'default',
    noRadius = false,
    backgroundUrl = '',
    backgroundOpacity = false,
    shadow = false,
    disabled = false,
    children,
    ...rest
  }) => (
    <S.Box
      type={type}
      kind={kind}
      noRadius={noRadius}
      backgroundUrl={backgroundUrl}
      backgroundOpacity={backgroundOpacity}
      shadow={shadow}
      disabled={disabled}
      {...rest}
    >
      {children}
    </S.Box>
  )
)

Box.displayName = 'Box'
