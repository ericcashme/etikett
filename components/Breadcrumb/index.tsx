import React, { FC, memo } from 'react'
import Link from 'next/link'
import { Link as LinkComp } from '../Link'
import * as S from './styles'
import { BreadcrumbProps, Page } from './types'

export const Breadcrumb: FC<BreadcrumbProps> = memo(({ pages, inverse = false }) => {
  const mutPages: Page[] = [...pages]
  const actualPage = mutPages.pop() as Page

  return (
    <S.Breadcrumb itemScope itemType="https://schema.org/BreadcrumbList">
      {mutPages.map((item, index) => (
        <S.BreadcrumbItem
          inverse={inverse}
          key={`bc-${item.path}`}
          itemProp="itemListElement"
          itemScope
          itemType="https://schema.org/ListItem"
        >
          <Link href={item.path} as={item.path} passHref>
            <LinkComp>
              <span itemProp="name">{item.label}</span>
              <meta itemProp="position" content={String(index + 1)} />
            </LinkComp>
          </Link>
        </S.BreadcrumbItem>
      ))}

      <S.BreadcrumbItem inverse={inverse} itemProp="itemListElement" itemScope itemType="https://schema.org/ListItem">
        <span itemProp="name">{actualPage.label}</span>
        <meta itemProp="position" content={String(mutPages.length + 1)} />
      </S.BreadcrumbItem>
    </S.Breadcrumb>
  )
})

Breadcrumb.displayName = 'Breadcrumb'
