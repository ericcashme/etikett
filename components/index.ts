import CCol from './Grid/Col'
import CRow from './Grid/Row'
import Provider, { tokens } from './theme'

export const theme = tokens
export const ComponentsProvider = Provider
export const Row = CRow
export const Col = CCol

export { Alert } from './Alert'
export { Avatar } from './Avatar'
export { Box } from './Box'
export { Breadcrumb } from './Breadcrumb'
export { Button } from './Button/Button'
export { Buttons } from './Button/Buttons'
export { CardAddress } from './CardAddress'
export { CardBlog } from './CardBlog'
export { CardProduct } from './CardProduct'
export { Checkbox } from './Checkbox'
export { Collapse } from './Collapse/Collapse'
export { CollapseItem } from './Collapse/CollapseItem'
export { CookieConsent } from './CookieConsent'
export { Icon } from './Icon'
export { NewIcon } from './Icon/NewIcon'
export { IconsList } from './Icon/IconsList'
export { Input } from './Input'
export { HorizontalScroll } from './HorizontalScroll'
export { Link } from './Link'
export { List } from './List/List'
export { ListItem } from './List/ListItem'
export { LinearGradient } from './LinearGradient'
export { Loader } from './Loader'
export { Logo } from './Logo'
export { Modal } from './Modal'
export { Placeholder } from './Placeholder'
export { RatingStar } from './Rating'
export { Select } from './Select'
export { SelectCard } from './SelectCard'
export { Tag } from './Tag'
export { Text } from './Text'
export { Textarea } from './Textarea'
export { Tooltip } from './Tooltip'
export { User } from './User'
