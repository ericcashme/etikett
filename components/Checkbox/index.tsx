import React, { FC, memo } from 'react'
import { Text } from 'components'
import * as S from './styles'
import { CheckboxProps } from './types'

export const Checkbox: FC<CheckboxProps> = memo(
  ({ kind = 'square', id, name, value, checked, onChange, label, description, ...rest }) => (
    <S.CheckboxContainer htmlFor={id} {...rest}>
      <S.HiddenInput type="checkbox" id={id} name={name} value={value} checked={checked} onClick={onChange} readOnly />
      <S.Shape kind={kind} checked={checked}>
        <S.InnerShape kind={kind} checked={checked} />
      </S.Shape>

      <S.Infos>
        <Text as="span" margin="0 0.25rem 0 0">
          {label}
        </Text>

        {description && (
          <Text as="span" color="primary" margin="0">
            {description}
          </Text>
        )}
      </S.Infos>
    </S.CheckboxContainer>
  )
)

Checkbox.displayName = 'Checkbox'
