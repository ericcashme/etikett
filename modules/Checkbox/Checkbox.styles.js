import styled from 'styled-components'
import { theme } from 'components'
const { breakpoint, space, borders, color, font } = theme

export const CheckboxStyle = styled.label`
  display: flex;
  justify-content: flex-start;
  align-items: center;
  margin-bottom: ${(p) => (p.margin ? p.margin : space.md)};
  cursor: pointer;
  @media (max-width: ${breakpoint.lg}) {
    width: 100%;
    margin-top: ${space.md};
  }
`

export const HiddenInput = styled.input`
  position: absolute;
  opacity: 0;
  cursor: pointer;
  height: 0;
  width: 0;
`

export const Shape = styled.span`
  position: relative;
  display: inline-block;
  margin-right: ${space.xs2};
  border: ${borders.sizes.sm} solid ${color.grayscale.gray};
  background-color: ${color.grayscale.white};
  cursor: pointer;
  user-select: none;
  transition: all 0.4s ease;

  ${(p) =>
    p.kind === 'square' &&
    `
    height: ${space.md};
    width: ${space.md};
    border-radius: ${borders.radius.sm};
  `}

  ${(p) =>
    p.kind === 'circle' &&
    `
    height: 1.625rem;
    width: 1.625rem;
    border-radius: 50%;
  `}
  ${(p) =>
    p.checked &&
    `
    border-color: ${color.primary.dark};
    background-color: ${color.primary.dark};
  `}
`

export const InnerShape = styled.i`
  display: block;
  position: absolute;

  ${(p) =>
    p.kind === 'square' &&
    `
    left: 0.3125rem;
    top: 0.0625rem;
    width: 0.3125rem;
    height: 0.625rem;
    border-style: solid;
    border-color: ${color.grayscale.white};
    border-width: 0 ${borders.sizes.md} ${borders.sizes.md} 0;
    transform: rotate(45deg);
  `}

  ${(p) =>
    p.kind === 'circle' &&
    `
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
    margin: auto;
    border-radius: 50%;
    background-color: ${color.grayscale.white};
    transition: all 0.2s ease;
`}

${(p) =>
    p.checked
      ? `
      // width: ${space.xs};
      // height: ${space.xs};
      opacity: 1;
    `
      : `
      width: 1.625rem;
      height: 1.625rem;
      opacity: 0;
    `}
`

export const Label = styled.p`
  margin-bottom: 0;

  ${(p) =>
    p.kind === 'square' &&
    `
    font-size: ${font.size.xsmall};
  `}

  ${(p) =>
    p.kind === 'circle' &&
    `
    font-size: ${font.size.root};
    font-weight: ${font.weight.bold};
    letter-spacing: -0.03125rem;
  `}
`
