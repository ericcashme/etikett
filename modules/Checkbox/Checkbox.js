import React, { memo } from 'react';

import PropTypes from 'prop-types';

import * as S from './Checkbox.styles';

const Checkbox = ({ kind, id, name, value, checked, onChange, label, readOnly, ...rest }) => (
  <S.CheckboxStyle kind={kind} htmlFor={id} {...rest}>
    <S.HiddenInput
      id={id}
      name={name}
      type="checkbox"
      value={value}
      checked={checked}
      onChange={onChange}
      readOnly={readOnly}
    />
    <S.Shape kind={kind} checked={checked}>
      <S.InnerShape kind={kind} checked={checked} />
    </S.Shape>

    {label && <S.Label kind={kind}>{label}</S.Label>}
  </S.CheckboxStyle>
);

Checkbox.propTypes = {
  kind: PropTypes.oneOf(['square', 'circle']),
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  name: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  checked: PropTypes.bool.isRequired,
  onChange: PropTypes.func,
  readOnly: PropTypes.bool,
  label: PropTypes.string,
};

Checkbox.defaultProps = {
  kind: 'square',
  onChange: null,
  readOnly: false,
  label: '',
};

export default memo(Checkbox);
