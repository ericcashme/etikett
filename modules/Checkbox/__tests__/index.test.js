import React from 'react'
import { ComponentsProvider } from 'components'
import { render } from '@testing-library/react'
import Checkbox from '../index'

const CheckboxComponent = (props) => (
  <ComponentsProvider>
    <Checkbox {...props} />
  </ComponentsProvider>
)

describe('<Checkbox />', () => {
  it('Expect to render Checkbox with default snapshot', () => {
    expect(
      render(<CheckboxComponent onChange={() => {}} id="checkbox" name="checkbox" value="checkbox" checked />)
    ).toMatchSnapshot()
  })

  it('Expect to render Checkbox with snapshot and optional props', () => {
    expect(
      render(
        <CheckboxComponent
          id="checkbox"
          name="checkbox"
          value="checkbox"
          checked
          onChange={() => {}}
          readOnly
          label="Label"
          kind="circle"
        />
      )
    ).toMatchSnapshot()
  })

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    render(
      <CheckboxComponent
        id="checkbox"
        name="checkbox"
        value="checkbox"
        checked
        onChange={() => {}}
        readOnly
        label="Label"
        kind="circle"
      />
    )
    expect(spy).not.toHaveBeenCalled()
  })
})
