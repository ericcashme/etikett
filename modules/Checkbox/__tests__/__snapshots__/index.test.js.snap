// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> Expect to render Checkbox with default snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 1rem;
  cursor: pointer;
}

.c1 {
  position: absolute;
  opacity: 0;
  cursor: pointer;
  height: 0;
  width: 0;
}

.c2 {
  position: relative;
  display: inline-block;
  margin-right: 0.25rem;
  border: 1px solid #8C9093;
  background-color: #ffffff;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all 0.4s ease;
  transition: all 0.4s ease;
  height: 1rem;
  width: 1rem;
  border-radius: 0.125rem;
  border-color: #007CCC;
  background-color: #007CCC;
}

.c3 {
  display: block;
  position: absolute;
  left: 0.3125rem;
  top: 0.0625rem;
  width: 0.3125rem;
  height: 0.625rem;
  border-style: solid;
  border-color: #ffffff;
  border-width: 0 2px 2px 0;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  opacity: 1;
}

@media (max-width:48rem) {
  .c0 {
    width: 100%;
    margin-top: 1rem;
  }
}

<body>
    <div>
      <label
        class="c0"
        for="checkbox"
        kind="square"
      >
        <input
          checked=""
          class="c1"
          id="checkbox"
          name="checkbox"
          type="checkbox"
          value="checkbox"
        />
        <span
          class="c2"
          kind="square"
        >
          <i
            class="c3"
            kind="square"
          />
        </span>
        
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="Checkboxstyles__CheckboxStyle-plj925-0 jySHSR"
      for="checkbox"
      kind="square"
    >
      <input
        checked=""
        class="Checkboxstyles__HiddenInput-plj925-1 jMtezP"
        id="checkbox"
        name="checkbox"
        type="checkbox"
        value="checkbox"
      />
      <span
        class="Checkboxstyles__Shape-plj925-2 krBcNB"
        kind="square"
      >
        <i
          class="Checkboxstyles__InnerShape-plj925-3 eThaRx"
          kind="square"
        />
      </span>
      
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Checkbox /> Expect to render Checkbox with snapshot and optional props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 1rem;
  cursor: pointer;
}

.c1 {
  position: absolute;
  opacity: 0;
  cursor: pointer;
  height: 0;
  width: 0;
}

.c2 {
  position: relative;
  display: inline-block;
  margin-right: 0.25rem;
  border: 1px solid #8C9093;
  background-color: #ffffff;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all 0.4s ease;
  transition: all 0.4s ease;
  height: 1.625rem;
  width: 1.625rem;
  border-radius: 50%;
  border-color: #007CCC;
  background-color: #007CCC;
}

.c3 {
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  margin: auto;
  border-radius: 50%;
  background-color: #ffffff;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  opacity: 1;
}

.c4 {
  margin-bottom: 0;
  font-family: undefined;
  font-size: 1rem;
  font-weight: bold;
  -webkit-letter-spacing: -0.03125rem;
  -moz-letter-spacing: -0.03125rem;
  -ms-letter-spacing: -0.03125rem;
  letter-spacing: -0.03125rem;
}

@media (max-width:48rem) {
  .c0 {
    width: 100%;
    margin-top: 1rem;
  }
}

<div>
      <label
        class="c0"
        for="checkbox"
        kind="circle"
      >
        <input
          checked=""
          class="c1"
          id="checkbox"
          name="checkbox"
          readonly=""
          type="checkbox"
          value="checkbox"
        />
        <span
          class="c2"
          kind="circle"
        >
          <i
            class="c3"
            kind="circle"
          />
        </span>
        <p
          class="c4"
          kind="circle"
        >
          Label
        </p>
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="Checkboxstyles__CheckboxStyle-plj925-0 jySHSR"
      for="checkbox"
      kind="circle"
    >
      <input
        checked=""
        class="Checkboxstyles__HiddenInput-plj925-1 jMtezP"
        id="checkbox"
        name="checkbox"
        readonly=""
        type="checkbox"
        value="checkbox"
      />
      <span
        class="Checkboxstyles__Shape-plj925-2 iPEHrr"
        kind="circle"
      >
        <i
          class="Checkboxstyles__InnerShape-plj925-3 ddmuBN"
          kind="circle"
        />
      </span>
      <p
        class="Checkboxstyles__Label-plj925-4 kGHTHO"
        kind="circle"
      >
        Label
      </p>
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
